### ISSUE TITLE
A few thoughts about row and column options

### ISSUE THREAD

**Issue by Werbschaft** [getkirby/blocks#44]
Ever since I first saw the layouts, I've been wondering what it might look like with the options for them and how to use them. Without these options the layout field would be useless at least for us and we would go back to the nested blocks. But currently these are not usable either and we don't know if this will be very popular or not at all in the first release. A small thought: Notion always had a to-do list, maybe it would be good to publish it again. Sometimes it is hard to follow the discussion on Discord, so here are some thoughts

We need options for the rows and the columns. The options for the rows could easily be inserted visually into the existing layout as a drop down.

<img width="412" alt="Bildschirmfoto 2020-11-13 um 11 27 19" src="https://user-images.githubusercontent.com/2178466/99062527-6dfb1300-25a3-11eb-83be-9b0e7c68f2d2.png">

Again, it would be great if you could work with tabs otherwise we would have 30+ fields in the drawer. Ideally you could set these options globally for the whole field. But how exactly will this look like, it is hard to imagine at the moment. Here is a possible example:

Simple Options
```
fields: 
  blocks:
    label: Text
    type: blocks
    rowOptions:
      fields:
        background:
          label: Background
          type: select
          options:
            "white": "White"
            "black": "Black"
```
Simple extend
```
fields: 
  blocks:
    label: Text
    type: blocks
    rowOptions: blocks/rowOptions
```

But what is much more difficult are the options for the individual columns. Where could the options be placed visually? I thought about this for a long time but nothing came up. **Is it even planned to have row AND column options?**

Whatever question we asked ourselves, if it is possible or useful to change the arrangement of the columns, for example to move column 1 to the 3rd position. However, this could also be very confusing and difficult to implement. Or even if it is possible to change the number of columns or the layout afterwards. Whereby I think that would be too much, but I wanted to mention it anyway.

What would be essential at least for us to start the block field:

- nested blocks
- Columns AND Rows Options
- more detailed documentation
- disable Block Preview for Standard Blocks

**Comment by bastianallgeier** [comment]
Thanks for all your input on this. 

Layout/row options are quite simple from a UI perspective and I will start working on them either today or on Monday. I'm not 100% decided on the yaml syntax for them though. I wonder if there should be just one global set of options for all layouts (as you suggested) or individual options depending on the layout. 

My initial idea was to have the option for more extensive layout settings. Something like this:

```yaml

**Comment by bastianallgeier** [comment]
Thanks for all your input on this. 

Layout/row options are quite simple from a UI perspective and I will start working on them either today or on Monday. I'm not 100% decided on the yaml syntax for them though. I wonder if there should be just one global set of options for all layouts (as you suggested) or individual options depending on the layout. 

My initial idea was to have the option for more extensive layout settings. Something like this:

```yaml
fields: 
  layout:
    label: Text
    type: layout
    layouts: 
      - columns: "1/2, 1/2"
        fields:            
          background:
            label: Background
            type: select
            options:
              "white": "White"
              "black": "Black"
      - columns: "1/4, 1/4, 1/4, 1,4"
        fields:            
          background:
            label: Background
            type: select
            options:
              "white": "White"
              "black": "Black"
```

This could get very complex, but it would probably also solve the problem of additional column options. You could easily inject additional column settings as fields in the layout setup instead of having even more drawers for each individual column. 

I also agree that it's super difficult to come up with a good UI for column settings. Where would they go? So this seems like my preferred path right now. 

It would also be possible to have global fields for all layouts and stick with our simple layout definition syntax, but if you want to be more specific you could switch to the more complex syntax. Not 100% sure yet. As a first step I will build a simple layout options drawer and then see how it could be configured.

**Comment by Werbschaft** [comment]
Gladly, I think a lot about the new field, to be honest, it occupies me almost day and night.

This is also a pretty good approach! In this example it would be important that the extending works properly, otherwise you could have a huge YAML code. We are generally very satisfied with the extending, you can outsource large parts of code, so that the actual field remains slim and visually clear. That every layout has its own options would not be necessary for us but maybe for someone else. I think you have to be careful not to follow current design trends but to think globally. For example, when the Windows 8 layout was still modern, it would be difficult to realize this only with rows of options.

What I am not sure about is if only the row should have options. I also can't really imagine that at the moment. I hope I understood you correctly, but I think it is currently planned that the whole row gets options, but not the columns. I think we could live with that as well!

It will only be difficult if for example the 2nd column needs special options. So the first column align-items: flex-start, but the 2nd column align-items: flex-end. Or a similar use case as described above with the Metro Layout.

**Comment by bastianallgeier** [comment]
I thought a bit more about it today and I will start with global options for all layouts. Otherwise I will never finish this thing :) I think it's still super powerful. 

With this you could also solve column settings at the moment via CSS by targeting specific columns in a layout. I.e. you could give the layout a certain class name and then via css define that the first column in that layout gets a different alignment than the others. I really think it's going to be powerful enough for the first version.

**Comment by Werbschaft** [comment]
I also think that is enough for a start! If you are interested, this was solved in the other pagebuilder we used at that time.

<img width="567" alt="Bildschirmfoto 2020-11-13 um 16 12 59" src="https://user-images.githubusercontent.com/2178466/99087672-55065800-25cb-11eb-89cd-3768721f3436.png">

**Comment by bastianallgeier** [comment]
✅
